File: BlackJack/BlackJack_csharp/BlackJack/BlackJack/model/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlackJack.model
{
	class Player
	{
		List<Card> m_hand;


		public Player()
		{
			m_hand = new List<Card>();
		}

		public void DealCard(Card a_card)
		{
			m_hand.Add(a_card);
		}

		public IEnumerable<Card> GetHand()
		{
			return (IEnumerable<Card>)m_hand.AsEnumerable();
		}

		public void ClearHand()
		{
			m_hand.Clear();
		}

		public void ShowHand()
		{

			foreach (Card c in GetHand())
			{
				c.Show();
			}
		}

		public int GetScoreOfHand()
		{
			int[] points = new int[] { 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10, 11 };	// Specific for black jack
			int score = 0;

			foreach (Card c in GetHand())
			{
				if (c.GetColor() != Card.Color.Hidden)
				{
					score += points[(int)c.GetValue()];
				}
			}

			if (score > 21)
			{
				foreach (Card c in GetHand())
				{
					if (c.GetValue() == Card.Value.Ace)
					{
						score -= 10;
						if (score <= 21)
						{
							break;
						}
					}
				}
			}


			return score;
		}
	}
}

File: BlackJack/BlackJack_java/model/Player.java
package model;

import java.util.LinkedList;

public class Player {

  LinkedList<Card> m_hand;
  
  public Player() {
    m_hand = new LinkedList<Card>();
  }
  
  public void clearHand() {
    m_hand = new LinkedList<Card>();
  }
  
  public void showHand() {
    for(Card c : getHand()){
      c.show();
    }
  }
  
  public void dealCard(Card a_c) {
    m_hand.add(a_c);
  }
  
  public Iterable<Card> getHand() {
    return m_hand;
  }
  
  public int getScore() {
    // the number of scores is dependant on the number of scorable values
    // as it seems there is no way to do this check at compile time in java ?!
    // cardScores[13] = {...};
    int cardScores[] = {
        2, 3, 4, 5, 6, 7, 8, 9, 10, 10 ,10 ,10, 11
    };
    assert (cardScores.length == Card.Value.Count.ordinal()) : "Card Scores array size does not match number of card values";
  
    
    int score = 0;

    for(Card c : getHand()) {
        if (c.getValue() != Card.Value.Hidden)
        {
            score += cardScores[c.getValue().ordinal()];
        }
    }

    if (score > 21)
    {
        for(Card c : getHand())
        {
            if (c.getValue() == Card.Value.Ace && score > 21)
            {
                score -= 10;
            }
        }
    }

    return score;
  }
  
}